api/v1alpha1/combination_types.go

```diff
@@ -101,6 +71,41 @@ type CombinationList struct {
 	Items           []Combination `json:"items"`
 }
 
+// GetCondition returns the condition with the provided type.
+func (c *Combination) GetCondition(condType string) *metav1.Condition {
```

Comment:
I recommend using the [apimachinery meta package](https://github.com/kubernetes/apimachinery/blob/e7b02e6514980d90b0df7364d3cb4c3316ace958/pkg/api/meta/conditions.go) that provides [a function](https://github.com/kubernetes/apimachinery/blob/e7b02e6514980d90b0df7364d3cb4c3316ace958/pkg/api/meta/conditions.go#L74) for getting conditions. Applies in all places where it makes sense.

Reply:
Good call, I replaced these two status functions with those defined by `apimachinery` with a thin wrapper to make interacting with `combination` statuses a bit cleaner.

```
// SetCondition sets the condition if it has not already been set
func (c *Combination) SetStatusCondition(condition metav1.Condition) {
	metautils.SetStatusCondition(&c.Status.Conditions, condition)
}
```